src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -156,26 +156,31 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
             {
                 if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
                 {
-                    var node = FindImplicitOrExplicitExpressionNode(context);
                     string action;
 
-                    // The formatting pass of our Default code action resolver rejects
-                    // implicit/explicit expressions. So if we're in an implicit expression,
-                    // we run the remapping resolver responsible for simply remapping
-                    // (without formatting) the resolved code action. We do not support
-                    // explicit expressions due to issues with the remapping methodology
-                    // risking document corruption.
-                    if (node is null)
+                    if (!TryGetOwner(context, out var owner))
                     {
-                        action = LanguageServerConstants.CodeActions.Default;
+                        // Failed to locate a valid owner for the light bulb
+                        continue;
                     }
-                    else if (node is CSharpImplicitExpressionSyntax)
+                    else if (IsSingleLineDirectiveNode(owner))
```

Comment:
Refactored the rest of the code to make this feel more natural; although this is the primary change that restricts us from showing the FQN directive when getting code actions for single line directives.